lib/engine/game/g_1824_cisleithania/game.rb

```diff
@@ -376,10 +381,11 @@ def special_handling_after_by_company(company, minor)
 
           # Need to handle construction railways when two player variant
           if pre_staatsbahn?(minor)
-            create_construction_railway_from_bought_pre_staatsbahn(company, minor)
+            handle_unreserve_of_pre_staatsbahn(company)
           else
-            create_construction_railways_from_coal_mine(company, minor)
+            create_bond_railway(company, minor)
```

Comment:
I don't like the structure of this code. This method is returning a boolean value to determine whether the last section of code in `after_buy_company` is executed, but you are also making changes to companies/corporations in the method. Can this block be moved to a separate method, called from `after_buy_company`?

Reply:
Have now refactored this a bit.